pkg/controller/configmapcainjector/configmap_injecting_controller.go

```diff
@@ -0,0 +1,178 @@
+	ic.configMapClient = configMapsClient
+	ic.syncHandler = ic.syncConfigMap
+
+	return ic
+}
+
+// Run begins watching and syncing.
+func (ic *ConfigMapCABundleInjectionController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer ic.queue.ShutDown()
+
+	// Wait for the stores to fill
+	if !cache.WaitForCacheSync(stopCh, ic.configMapHasSynced) {
+		return
+	}
+
+	glog.V(1).Infof("Starting workers")
+	for i := 0; i < workers; i++ {
+		go wait.Until(ic.runWorker, time.Second, stopCh)
+	}
+	<-stopCh
+	glog.V(1).Infof("Shutting down")
+}
+
+func (ic *ConfigMapCABundleInjectionController) enqueueConfigMap(obj interface{}) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+
+	ic.queue.Add(key)
+}
+
+func (ic *ConfigMapCABundleInjectionController) addConfigMap(obj interface{}) {
+	cm, ok := obj.(*v1.ConfigMap)
+	if !ok {
+		return
+	}
+
+	if !hasCABundleAnnotation(cm) {
+		return
+	}
+
+	glog.V(4).Infof("adding %s", cm.Name)
+	ic.enqueueConfigMap(cm)
+}
+
+func (ic *ConfigMapCABundleInjectionController) updateConfigMap(old, cur interface{}) {
+	cm, ok := cur.(*v1.ConfigMap)
+	if !ok {
+		return
+	}
+
+	if !hasCABundleAnnotation(cm) {
+		return
+	}
+
+	glog.V(4).Infof("updating %s", cm.Name)
+	ic.enqueueConfigMap(cm)
+}
+
+func (ic *ConfigMapCABundleInjectionController) runWorker() {
+	for ic.processNextWorkItem() {
+	}
+}
+
+// processNextWorkItem deals with one key off the queue.  It returns false when it's time to quit.
+func (ic *ConfigMapCABundleInjectionController) processNextWorkItem() bool {
+	key, quit := ic.queue.Get()
+	if quit {
+		return false
+	}
+	defer ic.queue.Done(key)
+
+	err := ic.syncHandler(key.(string))
+	if err == nil {
+		ic.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", key, err))
+	ic.queue.AddRateLimited(key)
+
+	return true
+}
+
+// syncSecret will sync the configMap with the CA.
+// This function is not meant to be invoked concurrently with the same key.
+func (ic *ConfigMapCABundleInjectionController) syncConfigMap(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	sharedConfigMap, err := ic.configMapLister.ConfigMaps(namespace).Get(name)
+	if kapierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// skip updating when the CA bundle is already there
+	if data, ok := sharedConfigMap.Data[InjectionDataKey]; ok {
+		if data == ic.ca {
+			return nil
+		}
+	}
+
+	configMapCopy := sharedConfigMap.DeepCopy()
+	// make a copy to avoid mutating cache state
+	configMapCopy.Data[InjectionDataKey] = ic.ca
+
+	_, err = ic.configMapClient.ConfigMaps(configMapCopy.Namespace).Update(configMapCopy)
+	return err
+}
+
+func hasCABundleAnnotation(cm *v1.ConfigMap) bool {
+	return cm.Annotations[InjectCABundleAnnotation] == "true"
```

Comment:
Do you think we should normalize a bit before comparing? Like trim and lowercase? I'd probably be frustrated if it wasn't working only to find out it's because I put `True` like in python

Reply:
good idea.